.circleci/config.yml

```diff
@@ -314,15 +314,84 @@ jobs:
       - run:
           name: Ensure non-editable mode works
           command: |
+              cd habitat-lab
               export PATH=$HOME/miniconda/bin:/usr/local/cuda/bin:$PATH
-              . activate habitat; cd habitat-lab
-              python -m pip install habitat-lab/
-              python -c "import habitat"
-              python -m pip install habitat-baselines/
-              python -c "import habitat_baselines"
+              conda create -y -n non-editable-install python=3.9
+              . activate non-editable-install
+              conda install -c aihabitat-nightly habitat-sim
+              pip install habitat-lab/
+              python -c 'import habitat; print("habitat version:", habitat.__version__)'
+              pip install habitat-baselines/
+              python -c 'import habitat_baselines; print("habitat_baselines version:", habitat_baselines.__version__)'
+
+      - run:
+          name: Ensure build and intall work
+          command: |
+              cd habitat-lab
+              export PATH=$HOME/miniconda/bin:/usr/local/cuda/bin:$PATH
+
+              conda create -y -n build-env python=3.9
+              . activate build-env
+              pip install --upgrade build
+              python -m build -s -w -C--global-option=egg_info -C--global-option=--tag-date habitat-lab/
+              python -m build -s -w -C--global-option=egg_info -C--global-option=--tag-date habitat-baselines/
+
+              conda create -y -n bdist-install python=3.9
+              . activate bdist-install
+              conda install -c aihabitat-nightly habitat-sim
+              conda create -n sdist-install --clone bdist-install
+
+              # install from built distribution:
+              . activate bdist-install
+              pip install habitat-lab/dist/habitat_lab*.whl
+              python -c 'import habitat; print("habitat version:", habitat.__version__)'
+              pip install habitat-baselines/dist/habitat_baselines*.whl
+              python -c 'import habitat_baselines; print("habitat_baselines version:", habitat_baselines.__version__)'
+
+              # install from source distribution:
+              . activate sdist-install
+              pip install habitat-lab/dist/habitat-lab*.tar.gz
+              python -c 'import habitat; print("habitat version:", habitat.__version__)'
+              pip install habitat-baselines/dist/habitat-baselines*.tar.gz
+              python -c 'import habitat_baselines; print("habitat_baselines version:", habitat_baselines.__version__)'
+
       - store_artifacts:
           path: habitat-lab/data/profile  # This is the benchmark profile
-
+  pypi_deploy:
+    <<: *gpu
+    parameters:
+      username_env_var:
+        description: "Twine username environment variable name"
+        type: env_var_name
+        default: TESTPYPI_USERNAME
```

Comment:
Why is this called TESTPYPI_xxx since it works for both, should it just be PYPI_xxx

Reply:
Yes, env var value is the same for testpypi and pypi. This is just an alias name to the same value. For TestPyPI I use env var names that start with "TEST" prefix, for PyPI without (because they are two different repositories even though we authenticate using API token and the username in this case should be set to "__token__").